tests/paste_events/paste_events.js

```diff
@@ -0,0 +1,112 @@
+/* bender-tags: editor,unit */
+/* bender-ckeditor-plugins: clipboard,autoid */
+/* bender-include: _helpers/pasting.js */
+/* global assertPasteEvent */
+
+'use strict';
+
+(function() {
+  bender.editor = {
+    config: {
+      enterMode: CKEDITOR.ENTER_P,
+      autoid: {
+        autostart: false
+      }
+    }
+  };
+
+  bender.test({
+
+    tearDown: function() {
+      this.editor.execCommand('autoid');
+    },
+
+    'test pasted heading id does not change when it is not a duplicate id': function() {
+      var bot = this.editorBot,
+        editor = bot.editor,
+        heading,
+        headings,
+        resumeAfter = bender.tools.resumeAfter,
+        headingWithId = '<h1 id="12345">This is a heading^</h1>',
+        headingWithDifferentId = '<h1 id="678910">This is also a heading</h1>';
+
+      bot.setHtmlWithSelection(headingWithId);
+
+      // hit enter key to prevent pasting text in same heading element
```

Comment:
That is strange.

Can you verify what the HTML looks like, after you press enter?